src/pages/index.js

```diff
@@ -106,31 +159,71 @@ function handleCardClick(cardData) {
   popupImage.open(cardData);
 }
 
-//* popupWithForm.js: Edit profile
+//* ==========================================
+//*       PopupWithForm: Edit profile
+//* ==========================================
 
 profileEditButton.addEventListener("click", () => {
-  const data = userInfo.getUserInfo();
+  let data = userInfo.getUserInfo();
   modalName.value = data.name;
   modalSubtitle.value = data.about;
   profileForm.open();
 });
 
 const profileForm = new PopupWithForm("#profile-edit-modal", (data) => {
-  userInfo.setUserInfo(data.name, data.subtitle);
-  profileForm.close();
+  profileForm.setSubmitText(true);
+  api
+    .editUserInfo(data)
+    .then((data) => {
+      userInfo.setUserInfo(data.name, data.about);
+      profileForm.close();
+    })
+    .finally(() => profileForm.setSubmitText(false));
 });
 profileForm.setEventListeners();
 
-//* popupWuthForm: Add Card
+//* ==========================================
+//*        PopupWithForm: Add Card
+//* ==========================================
 
 profileAddButton.addEventListener("click", () => {
   addFormValidator.resetValidation();
   addCardForm.open();
 });
 
 const addCardForm = new PopupWithForm("#profile-add-modal", (inputValues) => {
-  debugger;
-  cardSection.addItem(renderCard(inputValues));
-  addCardForm.close();
+  addCardForm.setSubmitText(true);
+  api
+    .addNewCard(inputValues)
+    .then((data) => {
+      cardSection.addItem(renderCard(data));
+      addCardForm.close();
+    })
+    .finally(() => addCardForm.setSubmitText(false));
 });
 addCardForm.setEventListeners();
+
+//* ==========================================
+//*     PopupWithForm: Delete Confirmation
+//* ==========================================
+
+const confirmation = new PopupWithConfirmation({
+  popupSelector: "#confirmation-modal",
+});
+confirmation.setEventListeners();
+
```

Comment:
Needs correction: don't forget to disable the avatar submit button when you open the avatar popup